acceptance_tests/utilities/pubsub_helper.py

```diff
@@ -53,10 +56,47 @@ def _ack_all_on_subscription(subscriber, subscription_path):
         return
 
     ack_ids = [message.ack_id for message in response.received_messages]
-
     if ack_ids:
         subscriber.acknowledge(subscription_path, ack_ids)
 
     # It's possible (though unlikely) that they could be > max_messages on the topic so keep deleting till empty
     if len(response.received_messages) == max_messages_per_attempt:
         _ack_all_on_subscription(subscriber, subscription_path)
+
+
+def _pull_exact_number_of_messages(subscriber, subscription_path, expected_msg_count, timeout):
+    # Synchronously pull messages one at at time until we either hit the expected number or the timeout passes.
+    messages = []
+    deadline = time.time() + timeout
+
+    while len(messages) < expected_msg_count and not time.time() > deadline:
+        # Why not just set the `max_messages` in the subscriber.pull call to the expected number? Because the
+        # subscriber client does not wait for the timeout before returning if it finds just at least one message,
+        # where we want to instead allow the full time for all the expected messages time to be published and pulled
+        response = subscriber.pull(subscription_path, return_immediately=True, max_messages=1)
```

Comment:
From the docs:

> Warning: setting this field to true is discouraged because it adversely impacts the performance of subscriptions.pull operations. We recommend that users do not set this field.

Happy to try it though, given it's just ATs and not actual prod code